lib/govukFrontendPaths.js

```diff
@@ -0,0 +1,53 @@
+// core dependencies
+const path = require('path')
+
+// npm dependencies
+const fse = require('fs-extra')
+
+/**
+ * GOV.UK Frontend paths (cached)
+ *
+ * @type {Map<string[], govukFrontendPaths>}
+ */
+const govukFrontendPathsCache = new Map()
+
+/**
+ * Find GOV.UK Frontend via search paths
+ *
+ * @param {string[]} searchPaths - Search paths for `require.resolve()`
+ * @returns {{ baseDir: string, includePath: string, assetPath: string, config: { [key: string]: unknown } }}
+ */
+function govukFrontendPaths (searchPaths = []) {
+  if (govukFrontendPathsCache.has(searchPaths)) {
```

Comment:
The cache isn't working, object keys are based on instance equality rather than having the same contents, this can be seen by running:

```
const map = new Map()
map.set(['a','b','c'], 'Hello')
map.get(['a','b','c']) // undefined
```

I'm aware that the old one had caching but I don't think it's necessary.  It runs a total of 5 times during startup, one is in the build phase and four are in the running app therefore caching would only save 3 calls and on one test on my laptop it took `0.46ms` to run.

Also, I've got [a generic function cache in another PR](https://github.com/alphagov/govuk-prototype-kit/pull/2328/files#diff-af24b81e74bfc6bd6f09c269fe1331823ac688caa0e2c484876761bce4ae2808R6-R28) which we can use if it would be helpful.  So I'd say that for this PR we should either remove caching (my preferred approach) or bring in the generic function cache which I can split into its own PR.

Reply:
✅ Done